client/verta/tests/registry/model_version/test_log_dataset.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from verta.dataset import Path
+
+pytestmark = pytest.mark.not_oss  # skip if run in oss setup. Applied to entire module
+
+class TestLogDataset:
+    def test_log_dataset(self, client, model_version, dataset):
+        key1, key2, key3 = "version1", "version2", "version3"
+
+        dataset_version1 = dataset.create_version(Path("conftest.py"))
+        dataset_version2 = dataset.create_version(Path(["modelapi_hypothesis/"]))
+        dataset_version3 = dataset.create_version(Path(["modelapi_hypothesis/"]))
+
+        model_version.log_dataset_version(key=key1, dataset_version=dataset_version1)
+        model_version.log_dataset_version(key=key2, dataset_version=dataset_version2)
+        model_version.log_dataset_version(key=key3, dataset_version=dataset_version3)
+
+        model_version = client.get_registered_model_version(id=model_version.id)
+
+        with pytest.raises(KeyError) as excinfo:
+            model_version.get_dataset_version("fake")
+
+        assert "no dataset found with key" in str(excinfo.value)
```

Comment:
fyi, we can do this if it's clearer!

```suggestion
        with pytest.raises(KeyError, match="no dataset found with key"):
            model_version.get_dataset_version("fake")
```